portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,47 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// the indexes recording the picture 
+// the slides in gallery.html is showing
+var slide1 = 0;
+var slide2 = 0;
+
 /**
- * Adds a random greeting to the page.
+ * Controls(flips) both slideshow in gallery.html
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function flip(/* num of pages to flip= */ num, 
+  /* index of the slide= */ sIndex) {
+  var slide;
+  if (sIndex == 1) {
+    slide1 += num;
+    slide = document.getElementById("slide1").getElementsByTagName("img");
```

Comment:
These two control structures are duplicates. Could be cleaned up by abstracting away the implementation into another function.